spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -80,6 +80,8 @@
 
 	private volatile Runnable stopCallback;
 
+	private volatile boolean readFromLeft = false;
```

Comment:
Well, quoting Redis.
>  BRPOP is identical to BLPOP with the only difference being that it pops elements from the tail of a list instead of popping from the head.

Something similar for the `RPUSH`:
> Insert all the specified values at the tail of the list stored at key.

I don't see reason to have the option as literal (`left/right`), because we have here only **boolean** state.
So, `popFromLeft `/`pushToRight` are sufficient options for our adapters, but I feel `tail="true/false"` should be enough simple and powerful option.